backend/flakestry/api/publish.py

```
@@ -46,18 +47,47 @@ def publish(
     # content={"message": "Private repositories are not supported, \
     # see https://github.com/flakestry/flakestry.dev/issues"})
 
+    github_headers = {
+        "Authorization": f"Bearer {github_token}",
+        "X-GitHub-Api-Version": "2022-11-28",
+        "Accept": "application/vnd.github+json",
+    }
+
+    owner_name, repository_name = token.repository.split("/")
+    if publish.ref:
+        ref = publish.ref
+    elif publish.version:
+        ref = f"refs/tags/{publish.version}"
+    else:
+        ref = "HEAD"
```

I'd throw an error here, there should be a ref passed.